numerics/combinatorics_test.cpp

```diff
@@ -0,0 +1,121 @@
+
+#include "numerics/combinatorics.hpp"
+
+#include "gtest/gtest.h"
+
+namespace principia {
+namespace numerics {
+
+TEST(Combinatorics, Binomial) {
+  // Pascal's triangle.
+  static_assert(Binomial(0, 0) == 1, "");
+
+  static_assert(Binomial(1, 0) == 1, "");
+  static_assert(Binomial(1, 1) == 1, "");
+
+  static_assert(Binomial(2, 0) == 1, "");
+  static_assert(Binomial(2, 1) == 2, "");
+  static_assert(Binomial(2, 2) == 1, "");
+
+  static_assert(Binomial(3, 0) == 1, "");
+  static_assert(Binomial(3, 1) == 3, "");
+  static_assert(Binomial(3, 2) == 3, "");
+  static_assert(Binomial(3, 3) == 1, "");
+
+  static_assert(Binomial(4, 0) == 1, "");
+  static_assert(Binomial(4, 1) == 4, "");
+  static_assert(Binomial(4, 2) == 6, "");
+  static_assert(Binomial(4, 3) == 4, "");
+  static_assert(Binomial(4, 4) == 1, "");
```

Comment:
We use C++17, so we don't empty strings.

Reply:
TIL.